mkosi/user.py

```diff
@@ -39,6 +39,33 @@ def name(cls) -> str:
     def home(cls) -> Path:
         return Path(f"~{cls.name()}").expanduser()
 
+    @classmethod
+    def is_regular_user(cls) -> bool:
+        return cls.uid >= 1000
+
+    @classmethod
+    def cache_dir(cls) -> Path:
+        if (env := os.getenv("XDG_CACHE_HOME") or (env := os.getenv("CACHE_DIRECTORY"))):
+            cache = Path(env)
+        elif (cls.is_regular_user() and Path.cwd().is_relative_to(INVOKING_USER.home())):
+            cache = INVOKING_USER.home() / ".cache"
+        else:
+            cache = Path("/var/cache")
```

Comment:
If I read this correctly, this means you have to provide `XDG_CACHE_HOME` or `CACHE_DIRECTORY` if you're invoking from outside of a user home and running unprivileged?

If this is intended, it should at least be mentioned in the documentation.

Reply:
@Cornelicorn I slightly changed the logic to use the home directory when we're not invoked as root, can you try again?